docs/channel.md

```diff
@@ -424,6 +424,47 @@ Y
 
 See also: [fromList](#fromlist) factory method.
 
+(channel-topic)=
+
+### topic
+
+:::{versionadded} 23.11.0-edge
+:::
+
+:::{note}
+This feature requires the `nextflow.preview.topic` feature flag to be enabled.
+:::
+
+The `topic` method is used to create a queue channel that receives the items from all process outputs in a particular "topic".
+
+A process output can be assigned to a topic using the `topic` option on an output:
+
+```groovy
+process foo {
+  output:
+  val('foo'), topic: 'my-topic'
+}
+
+process bar {
+  output:
+  val('bar'), topic: 'my-topic'
+}
+```
+
+Then, the `topic` method can be used to consume all items in the topic:
+
+```groovy
+Channel.topic('my-topic').view()
+```
+
+This approach is a convenient way to collect related items from many different sources without explicitly defining the channel logic (the `topic` method is essentially an implicit {ref}`operator-mix` operation). You can name topics however you want, and you can use different names to collect items for different "topics", as long as your process outputs and channel logic are consistent with each other.
+
+:::{warning}
+Any process that consumes a channel topic should not send any outputs to that topic, or else the pipeline will hang forever.
```

Comment:
What does it mean? A circular dependency?

Reply:
yes

Reply:
it should be possible to detect the condition and report an error

Reply:
I guess, if a process has a topic output, you could walk back up the DAG and check for topic factory calls. Because there could be operators between the topic factory and process. Can you think of anything simpler?

Reply:
Not sure sure, but while composing the dag or in the "normalization" phase it should be possible to identify it 

https://github.com/nextflow-io/nextflow/blob/topic-channel/modules/nextflow/src/main/groovy/nextflow/dag/DAG.groovy#L339-L340